### PR TITLE
Add envtest

### PR DESCRIPTION
**Pull request by sargun**
I know it's not an impressive test. It just brings up a node, and
makes sure it registers. Let's do more in the future.

### CODE REVIEW

File: scripts/validate/gomod.sh [hunk 1]
```diff
@@ -12,8 +12,6 @@ else
   echo "go mod tidy passed."
 fi
 
-git diff --exit-code go.mod go.sum || exit_code=$?
```
Comment (sargun) [inline scripts/validate/gomod.sh:15]: This has literally only provided headaches over the time. I don't see any good reason to have it.

File: node/node_test.go [hunk 2]
```diff
@@ -625,6 +625,7 @@ func TestManualConditionsPreserved(t *testing.T) {
 func testNode(t *testing.T) *corev1.Node {
 	n := &corev1.Node{}
 	n.Name = strings.ToLower(t.Name())
+	n.Name = strings.ReplaceAll(n.Name, "/", "-")
```
Comment (sargun) [inline node/node_test.go:628]: Turns out node names with `/` break the kubernetes.

**Comment by sargun** [comment]
@adrienjt We talked about this in your previous PR.

File: node/e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var enableEnvTest = flag.Bool("envtest", false, "Enable envtest based tests")
+
+func TestMain(m *testing.M) {
```
Comment (cpuguy83) [inline node/e2e_test.go:23]: Maybne these tests should go in their own package?

Comment (sargun) [reply node/e2e_test.go:23]: I wanted to do that / thought about doing that, but there's private members of pod controller that we want to prod at while running these tests.

Comment (cpuguy83) [reply node/e2e_test.go:23]: `<-node.chReady` seems like the only private member unless I missing something.
We should have a `Ready()` function for this just like pod controller.

Comment (sargun) [reply node/e2e_test.go:23]: Okay, I'll try to figure something out.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**